howso/direct/client.py

```diff
@@ -468,12 +485,14 @@ def _get_trainee_from_engine(self, trainee_id: str) -> Trainee:
         persistence = metadata.get('persistence', 'allow')
         trainee_meta = metadata.get('metadata')
         trainee_name = metadata.get('name')
+        transactional = metadata.get('transactional', False)
```

Comment:
Seems like storing this in the model itself could easily get out of sync with the true transactional state.

Reply:
At the Amalgam C API layer, there's no way to query this back.  The reverse of this is that there aren't that many ways to change the setting either: only the C `LoadEntity`, `CloneEntity`, and `StoreEntity` API calls can change it.  The corresponding `HowsoDirectClient.create_trainee()`, `copy_trainee(), and `persist_trainee()` calls all deal with the client-side metadata correctly.  The one other place that modifies the metadata state is `update_trainee()`, and that always preserves the current transactional setting.

I see the potential problem if something did manage to `self.amlg.load_entity()` without going through the client layer, but I'm not sure there's somewhere better to put the (per-process/per-Amalgam) value.